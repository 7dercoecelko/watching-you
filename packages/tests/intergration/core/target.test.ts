import { describe, expect, it } from 'vitest';
import { wait, mousemove } from '../../utils';
import { WatchingYou } from '../../watchingYou';
import { mockGetBoundingClientRectSpy } from '../../mocks';

mockGetBoundingClientRectSpy();

describe('Target', () => {
  it('It should be watching mouse', async () => {
    document.body.innerHTML = `
      <div id="watcher"></div>
    `;
    const watcherDom =
      document.querySelector<HTMLElement>('#watcher');
    if (watcherDom === null) throw Error('watcherDom is not found');
    const watchingYou = new WatchingYou(watcherDom);
    watchingYou.start();
    expect(watcherDom.style.transform).toMatchSnapshot();
    await mousemove(300, 300);
    expect(watcherDom.style.transform).toMatchSnapshot();
  });

  it('It should be watching dom', async () => {
    document.body.innerHTML = `
      <div id="watcher"></div>
      <div id="target">target</div>
    `;
    const watcherDom =
      document.querySelector<HTMLElement>('#watcher');
    const targetDom = document.querySelector<HTMLElement>('#target');
    if (watcherDom === null) throw Error('watcherDom is not found');
    if (targetDom === null) throw Error('targetDom is not found');
    const watchingYou = new WatchingYou(watcherDom, {
      target: targetDom,
      targetType: 'dom',
    });
    watchingYou.start();
    await wait(100);
    expect(watcherDom.style.transform).toMatchSnapshot();
  });

  it('It should be watching input', async () => {
    document.body.innerHTML = `
      <div id="watcher"></div>
      <input id="target"/>
    `;
    const watcherDom =
      document.querySelector<HTMLElement>('#watcher');
    const targetDom = document.querySelector<HTMLElement>('#target');
    if (watcherDom === null) throw Error('watcherDom is not found');
    if (targetDom === null) throw Error('targetDom is not found');
    const watchingYou = new WatchingYou(watcherDom, {
      target: targetDom,
      targetType: 'input',
    });
    watchingYou.start();
    expect(watcherDom.style.transform).toMatchSnapshot();
    targetDom.setAttribute('value', 'foo bar');
    await wait(100);
    expect(watcherDom.style.transform).toMatchSnapshot();
  });
});
