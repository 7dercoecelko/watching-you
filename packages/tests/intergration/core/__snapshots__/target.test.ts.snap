// Vitest Snapshot v1

exports[`Target > It should be watching dom 1`] = `"translate3d(38.7px,31.7px,0px) rotate(129.4deg)"`;

exports[`Target > It should be watching input 1`] = `"translate3d(44.9px,22.1px,0px) rotate(116.2deg)"`;

exports[`Target > It should be watching input 2`] = `"translate3d(45.4px,20.9px,0px) rotate(114.7deg)"`;

exports[`Target > It should be watching mouse 1`] = `"translate3d(0px,0px,0px) rotate(0deg)"`;

exports[`Target > It should be watching mouse 2`] = `"translate3d(37.2px,33.4px,0px) rotate(131.9deg)"`;
